### ISSUE TITLE
aws-opensearchservice: cannot disable logging

### ISSUE THREAD

**Issue by SigniantJoe** [aws/aws-cdk#29200]
### Describe the bug

Enabling logging on OpenSearch clusters will populate LogPublishingOptions in the output template, but disabling logging leaves LogPublishingOptions empty so the cluster will retain the previous logging config. I don't see any documentation saying that enabling logging is a one-way trip, so I think this is a bug.

Specifically, I'm talking about these logging props:
```
new es.Domain(scope, id, {
 ...
 logging: {
    slowSearchLogEnabled: false,
    appLogEnabled: false,
    slowIndexLogEnabled: false,
  },
})
```

### Expected Behavior

```
    "LogPublishingOptions": {
        "ES_APPLICATION_LOGS": {
            "Enabled": false
        },
        "SEARCH_SLOW_LOGS": {
            "Enabled": false
        },
        "INDEX_SLOW_LOGS": {
            "Enabled": false
        }
    },
```

### Current Behavior

```
    "LogPublishingOptions": {},
```

### Reproduction Steps

```
const { Stack, Duration } = require('aws-cdk-lib');
const es = require('aws-cdk-lib/aws-opensearchservice');

class PocStack extends Stack {
  constructor(scope, id, props) {
    super(scope, id, props);

    const cluster = new es.Domain(this, 'OpenSearchCluster', {
      domainName: 'test-opensearch-cluster',
      version: es.EngineVersion.OPENSEARCH_2_11,
      capacity: {
        dataNodes: 1,
        dataNodeInstanceType: "t4g.medium.search",
      },
      ebs: {
        volumeSize: 10,
      },
      logging: {
        appLogEnabled: true,
      }
    });
  }
}

module.exports = { PocStack }
```

### Possible Solution

If you don't need the current behaviour, then have it perform the expected behaviour from above.

If there is a good reason it works that way or you want backwards-compatibility, then maybe have logging props like appLogEnabled default to some non-boolean value (e.g. `-1`). If left default (not specified by user), then LogPublishingOptions can be empty. If the user explicitly specifies either true or false, then populate the LogPublishingOptions to enable or disable logging as appropriate.

### Additional Information/Context

An environment hit the limit on cloudwatch resource policies, so I'm working on a change related to the new (much appreciated) `suppressLogsResourcePolicy` prop. The way things are currently, I'll have to manually (*shudder*) disable logging on a cluster to clear up policy space before I can deploy the new multi-cluster logging resource policy.

### CDK CLI Version

2.128.0

### Node.js Version

v18.18.2

### OS

OSX

### Language

TypeScript

**Comment by pahud** [comment]
I am not sure if this is a bug from cloudformation but we should note this in the CDK doc. Thank you for the report.

**Comment by pahud** [comment]
@SigniantJoe 

Do you mean if I specify this way

```ts
new opensearch.Domain(this, 'Domain', {
	version: opensearch.EngineVersion.OPENSEARCH_2_11,
	logging: {
		slowSearchLogEnabled: false,
		appLogEnabled: false,
		slowIndexLogEnabled: false,
	},
})
```

it should synthesize to 

```json
"LogPublishingOptions": {
        "ES_APPLICATION_LOGS": {
            "Enabled": false
        },
        "SEARCH_SLOW_LOGS": {
            "Enabled": false
        },
        "INDEX_SLOW_LOGS": {
            "Enabled": false
        }
    },
```

but we are actually getting

```
"LogPublishingOptions": {},
```

If yes, I agree with you.

My question is - if I have enabled it like

```ts
new opensearch.Domain(this, 'Domain', {
	version: opensearch.EngineVersion.OPENSEARCH_2_11,
	logging: {
		slowSearchLogEnabled: true,
		appLogEnabled: true,
		slowIndexLogEnabled: true,
	},
})
```

Now I need to disable them and update to

```ts
new opensearch.Domain(this, 'Domain', {
	version: opensearch.EngineVersion.OPENSEARCH_2_11,
})
```

Will they be disabled from the enabled state? If not, I think https://github.com/aws/aws-cdk/pull/29202 is still relevant?

**Comment by pahud** [comment]
Reopen this issue as it's still relevant.

**Comment by SigniantJoe** [comment]
@pahud you are correct in your latest interpretation about getting empty logPublishingOptions when it should be populated, which means that CDK is unable to disable logging.

I actually made my own PR with unit tests for this, which I think explains the issue: https://github.com/aws/aws-cdk/pull/29205 It's currently stuck waiting on me to updating snapshots, which I'm looking into

**Comment by SigniantJoe** [comment]
@pahud Should I make a new issue since the bot closed this one again? Also can you advise on what to do with my PR that now passes all checks?
